controllers/storagecluster/storagecluster_controller.go

```diff
@@ -211,6 +211,31 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	crdPredicate := predicate.Funcs{
+		CreateFunc: func(e event.TypedCreateEvent[client.Object]) bool {
+			crdAvailable, keyExist := r.AvailableCrds[e.Object.GetName()]
```

Comment:
The reconciler knows exactly which CRDs it needs or does not need to react to. Why are we checking a non-static map?

Reply:
We don't need to trigger the reconciliation for the creation of all the CRDs, we already know which crds we are interested in hence we can have this check and avoid triggering the reconcile for other CRDs